src/flow/src/adapter/flownode_impl.rs

```diff
@@ -92,6 +94,33 @@ impl Flownode for FlowWorkerManager {
                     .map_err(to_meta_err)?;
                 Ok(Default::default())
             }
+            Some(flow_request::Body::Flush(FlushFlow {
+                flow_id: Some(flow_id),
+            })) => {
+                // TODO(discord9): impl individual flush
+                debug!("Starting to flush flow_id={:?}", flow_id);
+                // lock to make sure writes before flush are written to flow
+                let _flush_lock = self.flush_lock.write().await;
```

Comment:
I think flush should not wait for all insert requests to be executed before continuing, but only for the insert requests before flush to be executed.

Reply:
> I think flush should not wait for all insert requests to be executed before continuing, but only for the insert requests before flush to be executed.

this is tokio's Rwlock, I thought they are fair which means readers after writer wouldn't prevent writer to get their lock?
ref: https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html
> The priority policy of Tokio’s read-write lock is fair (or [write-preferring](https://en.wikipedia.org/wiki/Readers%E2%80%93writer_lock#Priority_policies)), in order to ensure that readers cannot starve writers. Fairness is ensured using a first-in, first-out queue for the tasks awaiting the lock; if a task that wishes to acquire the write lock is at the head of the queue, read locks will not be given out until the write lock has been released. This is in contrast to the Rust standard library’s std::sync::RwLock, where the priority policy is dependent on the operating system’s implementation.

Reply:
> The priority policy of Tokio’s read-write lock is fair (or [write-preferring](https://en.wikipedia.org/wiki/Readers%E2%80%93writer_lock#Priority_policies)), in order to ensure that readers cannot starve writers. Fairness is ensured using a first-in, first-out queue for the tasks awaiting the lock; if a task that wishes to acquire the write lock is at the head of the queue, read locks will not be given out until the write lock has been released.

from [doc](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html).

It's what I want.

Reply:
How about releasing the lock immediately so it won't block `handle_inserts`.